packages/@aws-cdk/aws-apigatewayv2/lib/common/domain-name.ts

```diff
@@ -139,4 +195,35 @@ export class DomainName extends Resource implements IDomainName {
       truststoreVersion: mtlsConfig.version,
     };
   }
+
+  /**
+   * Adds a configuration to a domain name. Properties like certificate, endpoint type and security policy can be set using this method.
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apigatewayv2-domainname-domainnameconfiguration.html
+   * @param props domain name properties to be set
+   */
+  public addEndpoint(props: DomainNameProps) : void {
+    const domainNameConfig: CfnDomainName.DomainNameConfigurationProperty = {
+      certificateArn: props.certificate.certificateArn,
+      certificateName: props.certificateName,
+      endpointType: props.endpointType ? props.endpointType?.toString() : 'REGIONAL',
+      ownershipVerificationCertificateArn: props.ownershipCertificate?.certificateArn,
+      securityPolicy: props.securityPolicy?.toString(),
+    };
+
+    if (this.isDuplicateEndpointType(domainNameConfig.endpointType)) {
+      throw new Error('no two domain name configurations should have the same endpointType');
+    }
+
+    this.domainNameConfigurations.push(domainNameConfig);
+  }
+
+  // validates whether the new domain name configuration has a unique endpoint or not
+  private isDuplicateEndpointType(endpointType: string | undefined) : boolean {
+    for (let config of this.domainNameConfigurations) {
+      if (endpointType && endpointType == config.endpointType) {
+        return true;
+      }
+    }
+    return false;
+  }
```

Comment:
If I'm reading this correctly, this will allow one endpoint type `undefined` and one endpoint type of `REGIONAL`. But this should not be allowed, right?

Can you fix this up and add a corresponding test case?

Reply:
It won't be that way. If you look at the `addEndpoint()` method which is the caller for this method, if there is no `EndpointType` defined by the customer, we set a default 'REGIONAL' endpoint. So this value will never be undefined. Some eslint code-check complained about it though, that's why I had to add `undefined` in the parameter.

Reply:
You can check the last test I have added. When I call `dn.addEndpoint()`, I am not giving an `EndpointType` in the input, but the code still figures out that I'm trying to add a REGIONAL endpoint here and throws an error.